src/target/riscv/riscv.c

```diff
@@ -480,10 +480,157 @@ static int riscv_create_target(struct target *target, Jim_Interp *interp)
 	return ERROR_OK;
 }
 
+static struct jim_nvp nvp_ebreak_config_opts[] = {
+	{ .name = "m", .value = RISCV_DCSR_EBREAK_M_CTL },
+	{ .name = "s", .value = RISCV_DCSR_EBREAK_S_CTL },
+	{ .name = "u", .value = RISCV_DCSR_EBREAK_U_CTL },
+	{ .name = "vs", .value = RISCV_DCSR_EBREAK_VS_CTL },
+	{ .name = "vu", .value = RISCV_DCSR_EBREAK_VU_CTL },
+	{ .name = NULL, .value = N_RISCV_DCSR_EBREAK_CTL }
+};
+
+#define RISCV_EBREAK_MODE_INVALID -1
+
+static struct jim_nvp nvp_ebreak_mode_opts[] = {
+	{ .name = "exception", .value = false },
+	{ .name = "halt", .value = true },
+	{ .name = NULL, .value = RISCV_EBREAK_MODE_INVALID }
+};
+
+static int jim_configure_ebreak(struct target *target, struct jim_getopt_info *goi)
+{
+	struct riscv_private_config * const pc = target->private_config;
+	struct jim_nvp *common_mode_nvp;
+	if (jim_nvp_name2value_obj(goi->interp, nvp_ebreak_mode_opts, goi->argv[0],
+				&common_mode_nvp) == JIM_OK) {
+		/* Here a common "ebreak" action is processed, e.g:
+		 * "riscv.cpu configure -ebreak halt"
+		 */
+		for (int ebreak_ctl_i = 0; ebreak_ctl_i < N_RISCV_DCSR_EBREAK_CTL; ++ebreak_ctl_i)
+			pc->dcsr_ebreak_ctls[ebreak_ctl_i] = common_mode_nvp->value;
+		return jim_getopt_obj(goi, NULL);
+	}
+
+	/* Here a "ebreak" action for a specific execution mode is processed, e.g:
+	 * "riscv.cpu configure -ebreak m halt"
+	 */
+	struct jim_nvp *ctrl_nvp;
+	if (jim_getopt_nvp(goi, nvp_ebreak_config_opts, &ctrl_nvp) != JIM_OK) {
+		jim_getopt_nvp_unknown(goi, nvp_ebreak_config_opts, /*hadprefix*/ true);
+		return JIM_ERR;
+	}
+	struct jim_nvp *mode_nvp;
+	if (jim_getopt_nvp(goi, nvp_ebreak_mode_opts, &mode_nvp) != JIM_OK) {
+		jim_getopt_nvp_unknown(goi, nvp_ebreak_mode_opts, /*hadprefix*/ true);
+		return JIM_ERR;
+	}
+	pc->dcsr_ebreak_ctls[ctrl_nvp->value] = mode_nvp->value;
+	return JIM_OK;
+}
```

Comment:
I believe it would be good to check the number of arguments in this function (`goi->argc`).

Here is an example how to do it: https://github.com/riscv-collab/riscv-openocd/blob/bfc12522395af86ba634aa8e085a8051ca6fd43c/src/target/arm_dap.c#L348-L351

Reply:
Addressed. Thanks for the catch!
Though I believe `Jim_WrongNumArgs()` is used incorrectly throughout the codebase. Please see https://review.openocd.org/c/openocd/+/8643 for details.